### PR TITLE
Add test for prover log regression

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Changes

Add a test that notifies us when we make changes that change anything about verification, with the caveat that the Dafny input program is not using many Dafny features, so verification changes in specific Dafny features might be missed.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyPipeline.Test/DafnyPipeline.Test.csproj [hunk 1]
```diff
@@ -43,5 +43,8 @@
 	  <Content Remove="_plugins/**" />
 	  <EmbeddedResource Include="_plugins/**" />
     <None Remove="_plugins\**" />
+    <None Update="expectedProverLog.txt">
```
Comment (atomb) [inline Source/DafnyPipeline.Test/DafnyPipeline.Test.csproj:46]: It's a tiny thing, but I might call this `expectedProverLog.smt2` or `SomeProgram.smt2`.

**Review by atomb — APPROVED** [review]
Thanks for doing this! I left one comment, but it's not critical.

File: Source/DafnyPipeline.Test/ProverLogStability.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  result := methodFormal == SomeFunc(42);
+}
+";
+
+    public ProverLogStabilityTest(ITestOutputHelper testOutputHelper) {
+      this.testOutputHelper = testOutputHelper;
+    }
+
+    [Fact]
+    public async Task ProverLogRegression() {
```
Comment (robin-aws) [inline Source/DafnyPipeline.Test/ProverLogStability.cs:68]: I think we need a sizable comment here to describe the purpose of the test. Something like "this test only provides data on when changes are made that change the encoding of Dafny programs to SMT programs. A failure is meant to provide this data, not block changes - if you are contributing a change that makes it fail...". Probably even with some background about the fundamental brittleness problem.

File: Source/DafnyPipeline.Test/ProverLogStability.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+namespace DafnyPipeline.Test {
+  // Main.Resolve has static shared state (TypeConstraint.ErrorsToBeReported for example)
+  // so we can't execute tests that use it in parallel.
+  [Collection("Singleton Test Collection - Resolution")]
+  public class ProverLogStabilityTest {
+
+    private readonly ITestOutputHelper testOutputHelper;
+
+    // All types of top level declarations.
+    readonly string originalProgram = @"
```
Comment (robin-aws) [inline Source/DafnyPipeline.Test/ProverLogStability.cs:22]: This is a good start at least, but I feel like it's likely one of the integration test files would cover more features, especially some non-trivial assertion verification.

**Review by robin-aws — CHANGES_REQUESTED** [review]
VERY glad to have this, thanks. Blocking on deeply commenting the purpose here though - otherwise I worry some future contributor will update it blindly or even delete it, thinking it's just a poorly-written brittle unit test.

**Review by robin-aws — APPROVED** [review]
Thanks for adding the comment :)

**Review by robin-aws — APPROVED** [review]
🚀

**Pull request merged**